prime-dotnet-webapi/Controllers/EnrolleePaperSubmissionsController.cs

```diff
@@ -258,5 +268,72 @@ public async Task<ActionResult> FinalizeEnrolleePaperSubmission(int enrolleeId)
 
             return Ok();
         }
+
+        // HEAD: api/Enrollees/potential-paper-enrollee/5
+        /// <summary>
+        /// Gets all paper enrollees and checks whether or not there is a match in dateOfBirth with the current logged on enrollee.
+        /// </summary>
+        [HttpHead("potential-paper-enrollee/{dateOfBirth}")]
+        [Authorize(Roles = Roles.TriageEnrollee + "," + Roles.PrimeEnrollee)]
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status404NotFound)]
+        public async Task<ActionResult> GetPotentialPaperEnrolleeReturneeStatus(DateTime dateOfBirth)
+        {
+            var result = await _enrolleeService.IsPotentialPaperEnrolleeReturnee(dateOfBirth);
+
+            if (!result)
+            {
+                return NotFound();
+            }
+
+            return Ok();
+        }
+
+        // POST: api/Enrollees/5/potential-paper-enrollee
+        /// <summary>
+        /// Creates a new Enrollee who may have a a previous paper enrolment.
+        /// </summary>
+        [HttpPost("{enrolleeId}/potential-paper-enrollee", Name = nameof(CreateLinkWithPotentialPaperEnrollee))]
+        [Authorize(Roles = Roles.TriageEnrollee + "," + Roles.PrimeEnrollee)]
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status201Created)]
+        public async Task<ActionResult> CreateLinkWithPotentialPaperEnrollee(int enrolleeId, EnrolleeLinkedEnrolmentViewModel payload)
+        {
+            await _enrolleeService.CreateInitialLink(enrolleeId, payload.UserProvidedGpid);
+            return Ok();
+        }
+
+        // PUT: api/Enrollees/5/linked-gpid
+        /// <summary>
+        /// Updates the paper enrolment gpid that the user provided
+        /// </summary>
+        [HttpPut("{Id}/linked-gpid", Name = nameof(UpdateGpidLinkToPaperEnrollee))]
+        [Authorize(Roles = Roles.TriageEnrollee + "," + Roles.PrimeEnrollee)]
```

Comment:
Whenever you make an endpoint available to PrimeEnrollees, you have to check their UserId against the record's UserId else you are letting any user look at / edit any other user's data.

Reply:
Fixed